extension/json/include/json_scan.hpp

```diff
@@ -235,6 +235,7 @@ struct JSONScanLocalState {
 	void ThrowObjectSizeError(const idx_t object_size);
 	void ThrowInvalidAtEndError();
 
+	//! Must hold the lock
```

Comment:
I think our current approach for requiring a lock in a protected/private method is to pass a `unique_lock` reference as a param. Like we do in `CatalogSet::CreateDefaultEntries(CatalogTransaction transaction, unique_lock<mutex> &lock);`

Reply:
The lock must be held for the code within the function. There are two types of uses for this. One where we grab the lock and call the function right after, and one where we've been holding on to the lock and then need to call the function.

I don't understand what using a `unique_lock` and passing it to the function would achieve here. We then would also need to add a parameter `bool i_already_have_the_lock` and then not lock it if this is `true`, or we would need to unlock and then call the function, which would grab the lock again.

Reply:
I think the idea is that if you should only call a function with the lock, making it impossible to call the function without the lock makes things a bit cleaner. I see that we do both throughout the code for protected/private methods: adding a comment that holding the lock is required and passing a unique_lock or lock_guard as an argument (sometimes in a wrapper class like `ClientContextLock`)

I have no strong opinion here, Im fine with merging it like this

Reply:
I agree with being safe, but I will leave this as it is. Thanks!